java/client/src/main/java/glide/api/models/commands/scan/ScanOptions.java

```diff
@@ -0,0 +1,60 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.scan;
+
+import java.util.ArrayList;
+import java.util.List;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * This base class represents the common set of optional arguments for the SCAN family of commands.
+ * Concrete implementations of this class are tied to specific SCAN commands (SCAN, HSCAN, SSCAN,
+ * and ZSCAN).
+ */
+@SuperBuilder
+public abstract class ScanOptions {
+    /** <code>MATCH</code> option string to include in the <code>SCAN</code> commands. */
+    public static final String MATCH_OPTION_STRING = "MATCH";
+
+    /** <code>COUNT</code> option string to include in the <code>SCAN</code> commands. */
+    public static final String COUNT_OPTION_STRING = "COUNT";
+
+    /**
+     * The match filter is applied to the result of the command and will only include strings that
+     * match the pattern specified. If the set, hash, or list is large enough for scan commands to
+     * return only a subset of the set, hash, or list, then there could be a case where the result is
+     * empty although there are items that match the pattern specified. This is due to the default
+     * <code>
+     * COUNT</code> being <code>10</code> which indicates that it will only fetch and match <code>10
```

Comment:
Fix the linebreak. YOu can also add `{@link #count}` or `{@link #count()}` - ensure that it is accessible for user, because `count` is a private field.

Reply:
Are you suggesting to replace all <code>COUNT</code> comments with {@link ScanOptionsBuilder#count()} references?

These comments are written about the protocol rather than the class itself so I'd be inclined to leave <code>COUNT</count> but am OK either way.

Note that we would need to reference ScanOptionsBuilder#count() since the ScanOptions itself doesn't have a publically accessible count() function.

Reply:
Adding `{@link` is optional, if it isn't tricky.

Reply:
I'll skip since it sort of mixes up code with protocol comments and also is a bit messy to link to lombok-generated code.

Reply:
Done.